airflow/operators/sql_branch_operator.py

```diff
@@ -0,0 +1,175 @@
+
+    :param parameters: (optional) the parameters to render the SQL query with.
+    :type parameters: mapping or iterable
+
+
+    """
+
+    template_fields = ("sql",)
+    template_ext = (".sql",)
+    ui_color = "#a22034"
+    ui_fgcolor = "#F7F7F7"
+
+    @apply_defaults
+    def __init__(
+        self,
+        sql: str,
+        follow_task_ids_if_true: List[str],
+        follow_task_ids_if_false: List[str],
+        conn_id: str = "default_conn_id",
+        database: Optional[str] = None,
+        parameters: Optional[Union[Mapping, Iterable]] = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.conn_id = conn_id
+        self.sql = sql
+        self.parameters = parameters
+        self.follow_task_ids_if_true = follow_task_ids_if_true
+        self.follow_task_ids_if_false = follow_task_ids_if_false
+        self.database = database
+        self._hook = None
+
+    def _get_hook(self):
+        self.log.debug("Get connection for %s", self.conn_id)
+        conn = BaseHook.get_connection(self.conn_id)
+
+        allowed_conn_type = {
+            "google_cloud_platform",
+            "jdbc",
+            "mssql",
+            "mysql",
+            "odbc",
+            "oracle",
+            "postgres",
+            "presto",
+            "sqlite",
+            "vertica",
+        }
+        if conn.conn_type not in allowed_conn_type:
+            raise AirflowException(
+                "The connection type is not supported by BranchSqlOperator. "
+                + "Supported connection types: {}".format(list(allowed_conn_type))
+            )
+
+        if not self._hook:
+            self._hook = conn.get_hook()
+            if self.database:
+                self._hook.schema = self.database
+
+        return self._hook
+
+    def execute(self, context: Dict):
+        # get supported hook
+        self._hook = self._get_hook()
+
+        if self._hook is None:
+            raise AirflowException(
+                "Failed to establish connection to '%s'" % self.conn_id
+            )
+
+        if self.follow_task_ids_if_true is None:
+            raise AirflowException(
+                "Expected task id or task ids assigned to follow_task_ids_if_true"
+            )
+
+        if self.follow_task_ids_if_false is None:
+            raise AirflowException(
+                "Expected task id or task ids assigned to follow_task_ids_if_false"
+            )
+
+        self.log.info(
+            "Executing: %s (with parameters %s) with connection: %s",
+            self.sql,
+            self.parameters,
+            self._hook,
+        )
+        record = self._hook.get_first(self.sql, self.parameters)
+        if not record:
+            raise AirflowException(
+                "No rows returned from sql query. Operator expected True or False return value."
+            )
+
+        if isinstance(record, list):
+            query_result = record[0][0]
```

Comment:
I get an error related to this line
```
  File "/home/airflow/dags/testbranch.py", line 126, in execute
    query_result = record[0][0]
TypeError: 'bool' object is not subscriptable
```

Did it work for you when you tested against other DBs?

Reply:
@eladkal  Thank you for review again. This is unexpected. I found an issue with the unit test and fixed this issue as well. The PR should be ready to review again.